experiment/adscert/signer.go

```diff
@@ -0,0 +1,129 @@
+package adscert
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"github.com/IABTechLab/adscert/pkg/adscert/api"
+	"github.com/IABTechLab/adscert/pkg/adscert/discovery"
+	"github.com/IABTechLab/adscert/pkg/adscert/signatory"
+	"github.com/benbjohnson/clock"
+	"github.com/prebid/prebid-server/config"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"time"
+)
+
+const SignHeader = "X-Ads-Cert-Auth"
+
+//Signer represents interface to access request Ads Cert signing functionality
+type Signer interface {
+	Sign(destinationURL string, body []byte) (string, error)
+}
+
+func NewAdCertsSigner(experimentAdCertsConfig config.ExperimentAdCerts) (Signer, error) {
+	if !experimentAdCertsConfig.Enabled {
+		return &NilSigner{}, nil
+	}
+	if len(experimentAdCertsConfig.InProcess.Origin) > 0 && len(experimentAdCertsConfig.Remote.Url) > 0 {
+		return nil, errors.New("both in-process and remote signers are specified. Please use just one signer")
+	}
+	if len(experimentAdCertsConfig.InProcess.Origin) > 0 {
+		//for initial implementation support in-process signer only
```

Comment:
Please add a unit test for this line.

Reply:
I tried to add init test for this, but had no luck - I cannot run it with fake data. Also it throws error that is not returned as a part of code and I cannot "catch" it. Not sure this test makes sense. I'm open to discuss possible test cases for instantiating in-process signer